pkg/network/setup/configstate.go

```diff
@@ -0,0 +1,110 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2023 Red Hat, Inc.
+ *
+ */
+
+package network
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	"kubevirt.io/client-go/log"
+
+	"kubevirt.io/kubevirt/pkg/network/cache"
+	neterrors "kubevirt.io/kubevirt/pkg/network/errors"
+)
+
+type ConfigState struct {
+	cacheCreator cacheCreator
+	vmiUID       string
+}
+
+func NewConfigState(cc cacheCreator, vmiUID string) ConfigState {
+	return ConfigState{cc, vmiUID}
+}
+
+func (c ConfigState) RunOnce(networkName string, prepareFunc func() error, configFunc func() error) error {
```

Comment:
The name `RunOnce` is a bit misleading. Only the `configFunc` is actually running once. The `prepareFunc` will be called again and again each time `RunOnce` is invoked.

Reply:
I renamed it to `Run` and added documentation to the method.